### PR TITLE
consensus, core, eth/downloader, params: 4844 chain validation

### PR DESCRIPTION
**Pull request by karalabe**
This PR implements validating the excessDataGa and dataGasUsed fields of Cancun headers according to the updated spec per https://github.com/ethereum/EIPs/pull/7062.

### CODE REVIEW

File: consensus/misc/eip4844.go [hunk 1]
```diff
@@ -27,13 +30,65 @@ var (
...
+		parentDataGasUsed   uint64
+	)
+	if parent.ExcessDataGas != nil {
+		parentExcessDataGas = *parent.ExcessDataGas
+		parentDataGasUsed = *parent.DataGasUsed
+	}
+	// Verify that the data gas limit remains within allowed bounds. To check
+	// the **actual** real usage we need to block body (transaction list), but
+	// at least do some sanity checks as early as possible.
+	if parentDataGasUsed > *header.DataGasUsed {
```
Comment (MariusVanDerWijden) [inline consensus/misc/eip4844.go:55]: Maybe we could deref the *header.DataGasUsed once?

Comment (karalabe) [reply consensus/misc/eip4844.go:55]: Ah, seems like such an insignificant optimisation. Maybe if it were in some super tight loop, but seems to just add code vs. make any dent.

File: consensus/misc/eip4844.go [hunk 2]
```diff
@@ -27,13 +30,65 @@ var (
...
+		if growth := parentDataGasUsed - *header.DataGasUsed; growth > params.BlobTxTargetDataGasPerBlock {
```
Comment (MariusVanDerWijden) [inline consensus/misc/eip4844.go:56]: Intuitively for me these cases are the wrong way around. You're calculating growth if parent > header and shrinkage if parent < header. I guess it depends on how you view it, but I kinda understood shrinkage as how much this header shrank compared to its parent

Comment (karalabe) [reply consensus/misc/eip4844.go:56]: indeed, will fix. my original code didn't have these named and then I started adding names and messed it up .

Comment (karalabe) [reply consensus/misc/eip4844.go]: Fixed.

**Review by MariusVanDerWijden — APPROVED** [review]
SGTM
One check that is implicit is that usedGas < excessDataGas +  params.MaxDataGasPerBlock. I've convinced myself that this check is not explicitly needed, but maybe I'm missing something

**Comment by karalabe** [comment]
usedGas < params.MaxDataGasPerBlock already, so the added check would be a bit funky?

**Pull request merged**